components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -5264,59 +5262,31 @@ public ResourceFile getWSDL(API api, String environmentName, String environmentT
-            resourcePath = APIUtil.getAPIPath((APIIdentifier) identifier);
-            identifierType = APIConstants.API_IDENTIFIER_TYPE;
-        } else if (identifier instanceof APIProductIdentifier) {
-            resourcePath = APIUtil.getAPIProductPath((APIProductIdentifier) identifier);
-            identifierType = APIConstants.API_PRODUCT_IDENTIFIER_TYPE;
-        }
-        Registry registry;
         try {
-            // Need user name with tenant domain to get correct domain name from
-            // MultitenantUtils.getTenantDomain(username)
-            String userNameWithTenantDomain = (userNameWithoutChange != null) ? userNameWithoutChange : username;
-            String apiTenantDomain = getTenantDomain(identifier);
-            int apiTenantId = getTenantManager().getTenantId(apiTenantDomain);
-            if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(apiTenantDomain)) {
-                APIUtil.loadTenantRegistry(apiTenantId);
-            }
-
-            if (this.tenantDomain == null || !this.tenantDomain.equals(apiTenantDomain)) { //cross tenant scenario
-                registry = getRegistryService().getGovernanceUserRegistry(
-                        getTenantAwareUsername(APIUtil.replaceEmailDomainBack(identifier.getProviderName())),
-                        apiTenantId);
-            } else {
-                registry = this.registry;
-            }
-            Resource resource = registry.get(resourcePath);
-            String accessControlProperty = resource.getProperty(APIConstants.ACCESS_CONTROL);
-            if (accessControlProperty == null || accessControlProperty.trim().isEmpty() || accessControlProperty
-                    .equalsIgnoreCase(APIConstants.NO_ACCESS_CONTROL)) {
-                if (log.isDebugEnabled()) {
-                    log.debug(identifierType + " in the path  " + resourcePath + " does not have any access control restriction");
+            if (identifier == null || !isAccessControlRestrictionEnabled) {
+                if (!isAccessControlRestrictionEnabled && log.isDebugEnabled() && identifier != null) {
+                    log.debug("Publisher access control restriction is not enabled. Hence the API/Product " +
+                            identifier.getName() + " should not be checked for further permission. Registry permission " +
+                            "check is sufficient");
                 }
                 return;
             }
-            if (APIUtil.hasPermission(userNameWithTenantDomain, APIConstants.Permissions.APIM_ADMIN)) {
+            String uuid  = identifier.getUUID();
+            String orgId = ApiMgtDAO.getInstance().getOrganizationIDByAPIUUID(uuid);
+            Organization org = new Organization(orgId);
+            DevPortalAPI devPortalApi = apiPersistenceInstance.getDevPortalAPI(org , uuid);
+            if (devPortalApi.getAccessControl().equalsIgnoreCase(APIConstants.NO_ACCESS_CONTROL)) {
```

Comment:
shall we switch the equality check?
```suggestion
            if (APIConstants.NO_ACCESS_CONTROL.equals(devPortalApi.getAccessControl()) {
```

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9997/commits/7b37b0df46826507c54b7038e3dc7d0b6037d0a2